export_snowflake/__init__.py

```diff
@@ -522,26 +528,126 @@ def flush_records(stream: str,
     db_sync.delete_from_stage(stream, s3_key)
 
 
+def transfer_data_in_parallel(s3_csv_lists, config, o, file_format_type):
+    parallel_transfer_start_time = time.time()
+    
+    # make the decision of how many processes to use later
+    pool = multiprocessing.Pool(4, initialize_s3_client_db_sync(config, o, file_format_type))
+    pool.map(download_file, s3_csv_lists)
+    pool.close()
+    pool.join()
+    
+    parallel_transfer_end_time = time.time()
+    stream = config["stream"]
+    LOGGER.info(f"Start time for transfering {stream} as {parallel_transfer_start_time} and end time as {parallel_transfer_end_time}")
+
+
+def download_file(job):
+    bucket, key, filename, stream = job
+
+    s3_client.download_file(bucket, key, filename)
+
+    # upload to snowflake
+    file_path = filename.replace("\\", "/")
+
+    s3_key = db_sync.put_to_stage(file_path, stream, temp_dir=None)
+
+    # remove from local after uploading to snowflake
+    os.remove(file_path)
+
+
+def export_to_snowflake(config, o, file_format_type): 
+    # retrieve the csv.gz lists and the schema file from the s3 bucket
+    s3_client = boto3.client('s3')
+
+    bucket = config["bucket"]
+    prefix = config["prefix"]
+    response = s3_client.list_objects_v2(Bucket=bucket, Prefix=prefix)
+
+    s3_csv_lists = []
+    schema_file = None
+    for obj in response['Contents']:
+        if obj['Key'].endswith('.csv.gz'):
```

Comment:
Does both python and spark version of parquet to csv step output in `gz` file?

Reply:
Yea, both python and spark are generating the compressed csv files for snowflake export